server/download.go

```diff
@@ -256,10 +256,15 @@ func (b *blobDownload) run(ctx context.Context, requestURL *url.URL, opts *regis
 				continue
 			}
 			defer resp.Body.Close()
-			if resp.StatusCode != http.StatusTemporaryRedirect && resp.StatusCode != http.StatusOK {
+
+			switch resp.StatusCode {
+			case http.StatusOK:
+				return requestURL, nil
+			case http.StatusTemporaryRedirect:
```

Comment:
Could we add more redirection status code such as 301 or 302?

Reply:
Added 301, 302, 303 and 308 to the check